aten/src/THC/generic/THCTensorMath.cu

```diff
@@ -256,6 +236,28 @@ void THCTensor_(catArray)(THCState *state, THCTensor *result,
   }
 }
 
+#if !defined(THC_REAL_IS_BOOL) /* non bool only part */
+
+void THCTensor_(zerosLike)(THCState *state, THCTensor *r_, THCTensor *input)
+{
+  THCAssertSameGPU(THCTensor_(checkGPU)(state, 2, r_, input));
+  THCTensor_(resizeAs)(state, r_, input);
+  THCTensor_(zero)(state, r_);
+}
+
+void THCTensor_(onesLike)(THCState *state, THCTensor *r_, THCTensor *input)
+{
+  THCAssertSameGPU(THCTensor_(checkGPU)(state, 2, r_, input));
+  THCTensor_(resizeAs)(state, r_, input);
+  THCTensor_(fill)(state, r_, ScalarConvert<int, scalar_t>::to(1));
+}
+
+ptrdiff_t
+THCTensor_(numel)(THCState *state, THCTensor *t)
```

Comment:
you could just define this on Bool -- seems perfectly reasonable.

Reply:
what do you mean? should i enable it for bool?

Reply:
sure.  It doesn't seem like it's exposed to python, so no harm, just internal functions can use it.